packages/@aws-cdk/aws-apigateway/lib/authorizer.ts

```diff
@@ -45,8 +45,10 @@ export abstract class CustomAuthorizer extends Resource implements IAuthorizer {
    * Typically, this is taken care of by the module while defining {@link Method}s via the {@link ResourceBase#addMethod | addMethod()} API call.
    *
    * @throws when a different RestApi is already associated to this authorizer.
+   *
+   * @internal
    */
-  public set restApi(restApi: IRestApi) {
+  public _bind(restApi: IRestApi) {
```

Comment:
As mentioned, this logic is not very useful. Make `bind` abstract and let the derived class store the restApi if it needs to.